error.go

```diff
@@ -2,11 +2,11 @@ package opensearchtools
 
 // Error encapsulates an error response from a given OpenSearch request.
 type Error struct {
-	RootCause    []Error `json:"root_cause"`
-	Type         string  `json:"type"`
-	Reason       string  `json:"reason"`
-	Index        string  `json:"index"`
-	ResourceID   string  `json:"resource.id"`
-	ResourceType string  `json:"resource.type"`
-	IndexUUID    string  `json:"index_uuid"`
+	RootCause    []*Error
```

Comment:
So, the top level `Error` on the response needs to be a pointer, so it can `omitempty` properly. I kind of cascaded that down so the methods could all return pointers rather than manually reference and deference in certain places. 

Would you prefer it not to be a slice of pointers?

Reply:
I was questioning whether it really needed to be a pointer to avoid unnecessary heap escaping.  I (incorrectly apparently) thought `omitempty` handled zero-valued structs as well. 😞